### PR TITLE
RC3.0: Make using API trimming optional

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
[AB#8158](https://dev.azure.com/fluidframework/internal/_workitems/edit/8158)

### CODE REVIEW

File: packages/service-clients/azure-client/package.json [hunk 1]
```diff
@@ -15,11 +15,11 @@
 	"exports": {
 		".": {
 			"import": {
-				"types": "./lib/public.d.ts",
+				"types": "./lib/legacy.d.ts",
```
Comment (tylerbutler) [inline packages/service-clients/azure-client/package.json:18]:
```suggestion
				"types": "./lib/index.d.ts",
```

File: packages/service-clients/azure-client/package.json [hunk 2]
```diff
@@ -15,11 +15,11 @@
...
 				"default": "./lib/index.js"
 			},
 			"require": {
-				"types": "./dist/public.d.ts",
+				"types": "./dist/legacy.d.ts",
```
Comment (tylerbutler) [inline packages/service-clients/azure-client/package.json:22]:
```suggestion
				"types": "./dist/index.d.ts",
```

File: packages/service-clients/azure-client/package.json [hunk 3]
```diff
@@ -55,7 +55,7 @@
 		}
 	},
 	"main": "lib/index.js",
-	"types": "lib/public.d.ts",
+	"types": "lib/legacy.d.ts",
```
Comment (tylerbutler) [inline packages/service-clients/azure-client/package.json:58]:
```suggestion
	"types": "lib/index.d.ts",
```

File: azure/packages/azure-service-utils/package.json [hunk 4]
```diff
@@ -15,21 +15,21 @@
...
-				"types": "./lib/legacy.d.ts",
+				"types": "./lib/index.d.ts",
 				"default": "./lib/index.js"
 			},
 			"require": {
-				"types": "./dist/legacy.d.ts",
+				"types": "./dist/index.d.ts",
 				"default": "./dist/index.js"
 			}
 		},
```
Comment (jason-ha) [inline azure/packages/azure-service-utils/package.json:35]: These need to stay point to legacy. Just the public (`"."`) and root `"types"` properties should be changed.

File: packages/service-clients/azure-client/package.json [hunk 5]
```diff
@@ -33,16 +33,6 @@
 				"default": "./dist/index.js"
 			}
 		},
-		"./legacy": {
-			"import": {
```
Comment (sonalideshpandemsft) [inline packages/service-clients/azure-client/package.json:37]: adding back the legacy exports here

File: packages/drivers/routerlicious-urlResolver/package.json [hunk 6]
```diff
@@ -14,12 +14,20 @@
 	"type": "module",
 	"exports": {
 		".": {
+			"trimmedAPI": {
+				"import": {
+					"types": "./lib/public.d.ts"
+				},
+				"require": {
+					"types": "./dist/pubic.d.ts"
```
Comment (jason-ha) [inline packages/drivers/routerlicious-urlResolver/package.json:22]: Typo. `pubic` to `public`

Comment (sonalideshpandemsft) [reply packages/drivers/routerlicious-urlResolver/package.json]: Thanks @jason-ha!!

**Review by jason-ha — APPROVED** [review]

**Pull request merged**